xdsl/transforms/convert_memref_to_ptr.py

```diff
@@ -228,37 +224,31 @@ def match_and_rewrite(self, op: func.CallOp, rewriter: PatternRewriter, /):
             return
 
         # rewrite arguments
-        insert_point = InsertPoint.before(op)
         new_arguments: list[SSAValue] = []
 
         # insert `memref -> ptr` casts for memref arguments values
         for argument in op.arguments:
             if isinstance(argument.type, memref.MemRefType):
-                rewriter.insert_op(
-                    cast_op := builtin.UnrealizedConversionCastOp.get(
-                        [argument], [ptr.PtrType()]
-                    ),
-                    insert_point,
+                rewriter.insert_op_before_matched_op(cast_op := ptr.ToPtrOp(argument))
+                new_arguments.append(cast_op.res)
+                cast_op.res.name_hint = argument.name_hint
+                argument.replace_by_if(
```

Comment:
What's with these additions, I can't really work out what they are doing

Reply:
Turns out that this one wasn't needed.